### PR TITLE
PWGMM: Add centrality, ...

### PR DESCRIPTION
**Pull request by joonsukbae**

### CODE REVIEW

File: PWGMM/Mult/Tasks/dndeta-hi.cxx [hunk 1]
```diff
@@ -318,551 +332,567 @@ struct MultiplicityCounter {
...
-            registry.fill(HIST("Tracks/ProcessMCCounting/hrecdndeta"), Double_t(kINELbg0), Double_t(kSel8), z, ttrack.mcParticle_as<Particles>().eta());
-            registry.fill(HIST("Tracks/ProcessMCCounting/hrecpt"), Double_t(kINELbg0), ttrack.mcParticle_as<Particles>().pt(), ttrack.pt());
-            registry.fill(HIST("Tracks/ProcessMCCounting/PhiEta"), Double_t(kINELbg0), ttrack.phi(), ttrack.eta());
-            registry.fill(HIST("Tracks/ProcessMCCounting/DCAXY"), Double_t(kINELbg0), ttrack.dcaXY());
-            registry.fill(HIST("Tracks/ProcessMCCounting/DCAZ"), Double_t(kINELbg0), ttrack.dcaZ());
+      auto collisionsample = collisions.sliceBy(perMcCol, mcCollision.globalIndex());
+      auto cent = -1;
+
+      if (collisionsample.size() != 1)
+        cent = -1;
```
Comment (aalkin) [inline PWGMM/Mult/Tasks/dndeta-hi.cxx:476]: Please add `{}` here after the condition.

File: PWGMM/Mult/Tasks/dndeta-hi.cxx [hunk 2]
```diff
@@ -318,551 +332,567 @@ struct MultiplicityCounter {
...
+    }
+    // cout<<"INEL? "<<bevtc[kINEL]<<"  INEL>0? "<<bevtc[kINELg0]<<endl;
+    for (auto ievtc = 1u; ievtc < kECend; ievtc++) {
+      if (bevtc[ievtc])
+        registry.fill(HIST("hgenzvtx"), Double_t(ievtc), genz, -1.0);
+    }
+    Int_t pid = 0;
+    for (auto& particle : mcParticles) {
+      auto p = pdg->GetParticle(particle.pdgCode());
+      if (std::abs(particle.pdgCode() == 310 && fabs(particle.eta()) < 0.5) && fabs(genz) < 10)
```
Comment (aalkin) [inline PWGMM/Mult/Tasks/dndeta-hi.cxx:772]: The parentheses of `std::abs` are closed incorrectly, it should be `std::abs(particle.pdgCode()) ==`.

File: PWGMM/Mult/Tasks/dndeta-hi.cxx [hunk 3]
```diff
@@ -318,551 +332,567 @@ struct MultiplicityCounter {
...
-  {
-    auto perCollisionMCSample = mcSample->sliceByCached(aod::mcparticle::mcCollisionId, mcCollision.globalIndex(), cache);
-    auto genz = mcCollision.posZ();
-    registry.fill(HIST("Tracks/ProcessGen/hgenzvtx"), Double_t(kINEL), genz);
-    for (auto& particle : perCollisionMCSample) {
-      auto p = pdg->GetParticle(particle.pdgCode());
-      if (p != nullptr) {
-        if (std::abs(p->Charge()) >= 3) {
-          registry.fill(HIST("Tracks/ProcessGen/hgendndeta"), Double_t(kINEL), genz, particle.eta());
+      auto mcCollision = collision.mcCollision();
```
Comment (aalkin) [inline PWGMM/Mult/Tasks/dndeta-hi.cxx:871]: This should be after `has_mcCollision()` is checked. And this variable is not actually used in the code below, is it a mistake?

**Review by aalkin — CHANGES_REQUESTED** [review]

File: PWGMM/Mult/Tasks/dndeta-hi.cxx (hunk 3 above)
Comment (joonsukbae) [reply PWGMM/Mult/Tasks/dndeta-hi.cxx]: I think it will be used later..

File: PWGMM/Mult/Tasks/dndeta-hi.cxx (hunk 2 above)
Comment (joonsukbae) [reply PWGMM/Mult/Tasks/dndeta-hi.cxx]: Yes, thanks for the exact advise

**Review by aalkin — APPROVED** [review]

**Pull request merged**